tasks/insights-client-unregistration.yml

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Check if insights-client is installed
+  ansible.builtin.package_facts:
+    manager: auto
+
+- name: Call insights status
+  command: insights-client --status
+  failed_when: false
```

Comment:
There is no case when `insights-client --status` will return an error that is unexpected or not able to be handled?
I guess the risk here is that status returns some strange value, and the `Unregister insights-client` task assumes that means the system is not registered, if `unregistered` and `NOT` are not part of the output

For example, if `insights-client --status` returns `0` if registered and `1` if not registered, then you could say 
```
  failed_when: __rhc_insights_status.rc not in [0, 1]
```
or you could use the stdout.  if the output is "System is registered" when registered and "System is unregistered" when unregistered, you could have a failed_when condition like this:
```yaml
  failed_when:
    - '"System is registered" not in __rhc_insights_status.stdout'
    - '"System is unregistered" not in __rhc_insights_status.stdout'
```

Reply:
I can limit the error code to be 0 or 1. Also I can change a bit the condition to check if `registered` is present in the output, to not assume it is registered when it returns a 1 and the `unregistered` and `NOT` are not part of the output.